Site Overview: Responsive layout

arthur791004 opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of DOTDASH-110, DOTCOM-13871

## Proposed Changes

* Make the site overview layout responsive

| Large | Small ~ Large | Small |
| - | - | - |
| <img width="772" height="933" alt="image" src="https://github.com/user-attachments/assets/4d81a07e-529b-4b7d-aecc-6c0302113dbc" /> | <img width="737" height="933" alt="image" src="https://github.com/user-attachments/assets/e2005498-8bbe-43f2-812a-4ca2d9ad79f5" /> | <img width="456" height="933" alt="image" src="https://github.com/user-attachments/assets/fe037f3b-60c7-4f94-9abf-0cfe14c48332" /> |

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

*

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to `/v2/sites/:site`
* Adjust the viewport width
* Ensure the layout looks good
  * `width >= 1080px`: 4 columns on the top and 2 columns on the bottom
  * `width < 782px`: 1 columns on both the top and the bottom, and the site preview is hidden
  * Others: 2 columns on the top and 1 columns on the bottom

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

arthur791004 committed:
Site Overview: Responsive layout

arthur791004 commented on client/dashboard/sites/overview/index.tsx:
```diff
@@ -22,9 +24,51 @@ import StorageCard from './storage-card';
+
+function SiteOverview( {
+	hideSitePreview = false,
+	breakpoints,
```
These options are intended for future backports

<img width="703" height="607" alt="image" src="https://github.com/user-attachments/assets/8ea3eacb-ea25-4029-be94-5a9f6ed200d7" />

taipeicoder replied:
Not necessarily for this PR, but we should also implement a hideHeader prop since the existing dashboard has its own header.

arthur791004 replied:
I'm not sure if it makes sense to support the `hideHeader` prop just for the sake of the backport. We could simply hide it using CSS instead. The same applies to `hideSitePreview`, though I’m adding it here since hiding that via CSS might be a bit trickier. What do you think?

taipeicoder replied:
Sounds good 👍

matt-west reviewed (commented):
I think we need to break a little earlier so the cards remain readable once they have content.

Let's drop to the tablet layout at 1080px, which should be the `$break-xlarge` breakpoint from Gutenberg. The mobile breakpoint is 782px (`$break-medium`).

We may need to tweak these again once all the cards have been implemented.

arthur791004 committed:
Adjust breakpoints

arthur791004 committed:
Add background color to SitePreviewCard when loading the iframe

p-jackson commented on client/dashboard/sites/overview/index.tsx:
```diff
@@ -47,10 +95,10 @@ function SiteOverview() {
-					<SitePreviewCard site={ site } />
-					<VStack className="site-overview-cards" spacing={ 6 }>
+			<VStack alignment="stretch" spacing={ isSmallViewport ? 5 : 10 }>
+				<Grid { ...gridLayout } gap={ spacing }>
```
If the `<Grid>` is safe for use then we should add it to the root `.eslintrc` to fix the import warning https://github.com/Automattic/wp-calypso/blob/8afca757bdaf3c66e0b21d75d3203362603d15b2/.eslintrc.js#L508

p-jackson commented on client/dashboard/sites/overview/index.tsx:
```diff
@@ -75,12 +123,21 @@ function SiteOverview() {
-					<VStack spacing={ 6 } justify="start">
+				<Divider
+					orientation="horizontal"
+					style={ { width: '100%', color: 'var(--wp-components-color-gray-100)' } }
```
The width style isn't necessary here, it already goes full width.

I also think the name of the colour variable is wrong. Afaict the idea of the `<Divider>` is that it has no inherent colour, and so it is fully customisable. So the fact we're explicitly setting a `color` makes sense. But I don't think we should name it after one of the component library colours. It's really part of the dashboard theme IMO.

```suggestion
					style={ { color: 'var(--dashboard__divider-color)' } }
```

In `style.scss` you can set `--dashboard__divider-color: var(--wp-components-color-gray-100, #f0f0f0);` if you want. But it seems to me, like the background colour, the colour is fully decided by our app, and so therefore it should be a `--dashboard__` variable.

p-jackson commented on client/dashboard/sites/overview/site-preview-card.tsx:
```diff
@@ -36,6 +36,7 @@ const SitePreviewCard = ( { site }: { site: Site } ) => {
 						bottom: 0,
 						left: 0,
 						overflow: 'hidden',
+						backgroundColor: 'var(--wp-components-color-gray-100)',
```
Same here, this feels like it should be a `--dashboard__` variable.

If you don't think it should be customisable per-product then you could always put it in `app/style.scss` instead of `app-dotcom/style.scss`. But it seems like it belongs in the same place as `--dashboard__background-color`.

p-jackson commented on client/dashboard/sites/overview/index.tsx:
```diff
@@ -22,9 +24,55 @@ import StorageCard from './storage-card';
+	isSmallViewport,
+}: {
+	count: number;
+	isLargeViewport: boolean;
+	isSmallViewport: boolean;
```
I feel like these should be combined into

```suggestion
	viewport: 'large' | 'medium' | 'small';
```

But I guess that's not how `useViewportMatch()` works 😢

arthur791004 replied:
Yeah, let's leave it as it is.

p-jackson reviewed (commented):
The changes test well to me 👍

What I find frustrating about `<Grid>` is that since the layout is passed as JS props, we have to do all the breakpoint calculations in JS. It could have been a pure CSS solution if it weren't for `<Grid>` 😮‍💨

taipeicoder commented:
A couple of things I noticed:

1. SiteFields drops down to a new line even though there's enough space.
<img width="1409" height="476" alt="SCR-20250716-inua" src="https://github.com/user-attachments/assets/9378bd56-6363-49a4-9e6c-b73e6cf6354a" />

2. Bottom left/right region is not guaranteed to have equal width. This is a pre-existing issue though.
<img width="1398" height="796" alt="Screenshot 2025-07-16 at 9 09 43 AM" src="https://github.com/user-attachments/assets/1825c97c-93be-47cd-845e-f34e7bafde48" />

arthur791004 committed:
Fix unbalanced card width

arthur791004 committed:
Add __experimentalGrid to eslint

arthur791004 committed:
Update css variable usage

arthur791004 committed:
Fix the site-overview-fields

arthur791004 committed:
Adjust the font size of heading

arthur791004 commented:
> SiteFields drops down to a new line even though there's enough space.

Fixed by https://github.com/Automattic/wp-calypso/pull/104748/commits/2c87d1d2de89de4cd7d3c5a33e355fc9274ea0d8

> Bottom left/right region is not guaranteed to have equal width. This is a pre-existing issue though.

Fixed by https://github.com/Automattic/wp-calypso/pull/104748/commits/9f0e9b5d5d14790ae09ad65c397744ba24b1f46a

arthur791004 committed:
Adjust the padding of the page-layout

arthur791004 committed:
Adjust the line-height of heading

fushar commented on client/dashboard/sites/overview/index.tsx:
```diff
@@ -23,35 +25,83 @@ import StorageCard from './storage-card';
+	} else if ( isSmallViewport ) {
+		columns = 1;
+	} else {
+		columns = Math.min( count / 2 );
```
```suggestion
		columns = Math.min( count, 2 );
```

?

arthur791004 replied:
Oops... Updated by https://github.com/Automattic/wp-calypso/pull/104748/commits/1ab331680671434ef7b037cd4e7e9b9fbd9cc8bc. It should be 1 column, 2 columns, and full columns (full = # of count).

arthur791004 committed:
Fix grid layout calculation

fushar commented on client/dashboard/sites/overview/index.tsx:
```diff
@@ -23,35 +25,88 @@ import StorageCard from './storage-card';
+
+	return {
+		columns: 2,
+		rows: Math.floor( count / 2 ) + ( count % 2 ),
```
Super nit: it's just `Math.ceil(count / 2)`

arthur791004 committed:
Simplify the condition

taipeicoder reviewed (approved):
Nicely done 👍 
I noticed that site title can overlap with the button, but we can address it later as we plan to add more buttons + ellipsis menu in the header.
 
<img width="376" height="779" alt="SCR-20250716-nlvt" src="https://github.com/user-attachments/assets/a25901f0-1bdf-45ff-b805-d46350daffb4" />
